libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -323,7 +323,11 @@ bool ASTJsonConverter::visit(FunctionDefinition const& _node)
 {
 	std::vector<pair<string, Json::Value>> attributes = {
 		make_pair("name", _node.name()),
-		make_pair(m_legacy ? "constant" : "isDeclaredConst", _node.stateMutability() == StateMutability::View),
+		make_pair(
+			m_legacy ? "constant" : "isDeclaredConst",
+			_node.stateMutability() == StateMutability::View ||
+			_node.stateMutability() == StateMutability::Pure
```

Comment:
Actually I don' think this should contain pure as it is only for the AST. By using this AST one couldn't reliable rebuild the source.